src/lightly_train/_data/mask_semantic_segmentation_dataset.py

```diff
@@ -105,16 +106,29 @@ def get_class_mapping(self) -> dict[int, int]:
         else:
             ignore_classes = self.args.ignore_classes
 
-        # Iterate over the classes and populate the class_mapppings.
+        # Iterate over the classes and populate the class_mappings.
         class_mapping = {}
-        class_counter = 0
-        for original_class in original_classes:
-            if original_class not in ignore_classes:
-                # Re-map the class.
-                class_mapping[original_class] = class_counter
-
-                # Update the class counter.
-                class_counter += 1
+        training_class_counter = 0
+
+        for class_key, class_info in input_classes.items():
+            # Skip if the class key itself is in ignore_classes
+            if class_key in ignore_classes:
+                continue
+
+            # Check if this ClassInfo has any non-ignored values
+            has_included_classes = any(
+                val not in ignore_classes for val in class_info.values
+            )
```

Comment:
I don't think this check is needed. `ignore_classes` only includes class keys but not the values. We shouldn't mix the two.

Maybe the mapping can even be written as:
```
class_mapping = {class_id: i for i, class_id in enumerate(self.args.included_classes())}
```